compiler/src/lnntoamm/Expr.ts

```diff
@@ -565,80 +590,62 @@ class Call extends Expr {
     }
     // first reduction
     const argTys = args.map((arg) => arg.ty);
-    const selFns = FunctionType.matrixSelect(fns, argTys, metadata.scope);
-    fns = selFns.map((selFn) => selFn[0]);
-    const retTy = Type.oneOf(selFns.map(([_fn, ty]) => ty));
+    const retTy = Type.generate();
+    const [selFns, selPTys, selRetTys] = FunctionType.matrixSelect(
+      fns,
+      argTys,
+      retTy,
+      metadata.scope,
+    );
+    fns = selFns;
+    // console.log('initial select for', ast.t.trim(), '->', fns);
+    // stdout.write('-> args ');
+    // console.dir(args, { depth: 4 });
```

Comment:
```suggestion
```

Reply:
this is useful for debugging so I was planning on keeping it